src/libraries/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/XmlWriterTraceListener.cs

```diff
@@ -258,8 +258,15 @@ private void WriteEndHeader()
             string? processName = s_processName;
             if (processName is null)
             {
-                using Process process = Process.GetCurrentProcess();
-                s_processName = processName = process.ProcessName;
+                try
+                {
+                    using Process process = Process.GetCurrentProcess();
+                    s_processName = processName = process.ProcessName;
+                }
+                catch (PlatformNotSupportedException)
```

Comment:
```suggestion
                catch (PlatformNotSupportedException) // Process isn't supported on Browser
```

Reply:
Did you consider instead doing a platform check?
```C#
if (RuntimeInformation.IsOSPlatform(OSPlatform.Browser) { ... }
else { ... }
```

Reply:
yeah but I think over the last couple PRs we landed on the consensus to prefer catching PNSE instead.

Reply:
> over the last couple PRs

Oh there have been more of these?  I must have missed that.  Could you point me to some examples?

Reply:
E.g. https://github.com/dotnet/runtime/commit/b62b16acf506f4e5c0d8b4ebaf18875ac9ad3477#diff-ce3739bb88265fab85205217d762f650R70-R79

Reply:
Ok.